src/Domain.LinnApps/Reports/ShortagesReportService.cs

```diff
@@ -0,0 +1,73 @@
+﻿namespace Linn.Purchasing.Domain.LinnApps.Reports
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using Linn.Common.Persistence;
+    using Linn.Common.Reporting.Models;
+    using Linn.Purchasing.Domain.LinnApps.Reports.Models;
+
+    public class ShortagesReportService : IShortagesReportService
+    {
+        private readonly IRepository<ShortagesEntry, string> shortagesEntryView;
+
+        public ShortagesReportService(
+            IRepository<ShortagesEntry, string> shortagesEntryView)
+        {
+            this.shortagesEntryView = shortagesEntryView;
+        }
+
+        public IEnumerable<ResultsModel> GetReport(
+            int purchaseLevel,
+            string vendorManager)
+        {
+            
+            var results = this.shortagesEntryView.FilterBy(x => (Convert.ToInt32(x.PurchaseLevel) <= purchaseLevel) &&
```

Comment:
do we need this Convert.ToInt32? (Not saying we don't - I just don't think I've had to do this before)

Reply:
This was something I wasn't entirely sure on. I had tried to Parse/TryParse this value but LINQ just wasn't having it - only ConvertToInt32 was valid. If I can change the field to an int I could do away with it.

Reply:
Ah I see... It really feels like the PURCH_LEVEL column should be a number and not a varchar. Andrew or Richard might know a reason why it isn't? It's PURCH_LEVEL on V_SHORT_TRIGGER_LEVELS.

you could force it to a number in your view definition? Or just leave it a string. If you leave it a string this code is fine

Reply:
I don't know why it's a string but it's definitely set as a string in the view so I wouldn't ever not present it as a string

Reply:
But you should still change it to be an int in your resource and drop the parse in your facade service.
right now it goes from:
an int from your options form -> to a string in your resource -> parsed back to an int in your facade -> passed as an int to your domain  
which is lots of pointless conversions

Reply:
Should just be an int all the way imo, and then this line 26 here will deal with the database type conversions (if you don't change your view to make the column a number). Hope that makes sense

Reply:
@richardiphillips we wrote the view for this report... could we just convert it to a number in the view sql? It feels like it really should be a number to me but I agree the C# should reflect the view properly

Reply:
ah ok. yeah i agree if you wrote the view for this report then just make it a number